code/UMLModel.py

```diff
@@ -1,102 +1,105 @@
-    
-    # **Write Documentation Here**
-    def rename_attribute(self):
-        print ("To be implemented")
-
-    ######################################################################
-    
-    # **Write Documentation Here**
-    def delete_attribute(self):
-        print ("To be implemented")
-
-    ######################################################################
-    
-    # **Write Documentation Here**
-    def create_relationship(self):
-        print ("To be implemented")
-
-    ######################################################################
-    
-    # **Write Documentation Here**
-    def delete_relationship(self):
-        print ("To be implemented")
-
-    ######################################################################
-    
-    # **Write Documentation Here**
-    def save_model(self):
-        print ("To be implemented")
-
-    ######################################################################
-    
-    # **Write Documentation Here**
-    def load_model(self):
-        print ("To be implemented")
-
-    ######################################################################
-
-    # **Write Documentation Here**
-    def list_class(self):
-        print ("To be implemented")
-
-    ######################################################################
-
-    # **Write Documentation Here**
-    def list_relationships(self):
-        print ("To be implemented")
-
-
-##########################################################################
-
+# A representation of the state of a UML Model
+# Description:     
+#   This file keeps track of all the classes that are in the UML Model
+#   It also handles saving/loading the Model to/from a JSON file 
+# Course:   CSCI 420 - Software Engineering
+# Authors:  Adisa, Amy, Carli, David, Joan
+# Date:     September 8 2020
+
+##########################################################################
+# Imports
+
+import UMLClass
+import UMLRelationship
+
+##########################################################################
+
+# Representation of a UML Model
+class UMLModel:
+
+    ######################################################################
+    
+    # Constructs an empty UML Model
+    def __init__(self):
+        
+        # Dictionary mapping class names to their class object
+        self.classes = {}
+
+    ######################################################################
+    
+    # **Write Documentation Here**
+    def create_class(self):
+        print ("To be implemented")
+
+    ######################################################################
+    
+    # **Write Documentation Here**
+    def rename_class(self):
+        print ("To be implemented")
+
+    ######################################################################
+    
+    # Deletes a given class
+    def delete_class(self, name:str):
+        if name in self.classes:
+            del self.classes[name]

```

Comment:
You should probably make sure you remove the dictionary entry as well. You could use the remove method for dictionaries.

Reply:
When you say dictionary entry, you're talking about one specific key-value pair entry, right? With del self.classes[name], when I tested it, it deleted both the key and its value, so the key-value pair is completely removed from the dictionary. That's what we want, right? Maybe I'm just misunderstanding.

Also, I will copy/paste the delete_class function into the create_class branch then! I'm not sure why it's saying that all of the lines were modified! Sorry about that, and thanks for saying something!

Reply:
Oh sorry! I've honestly never used del before. You're right that does delete the entry. That is a very questionable syntax on Python's part, but I guess that's all good then.

Reply:
If you prefer that I use the remove method though, I really don't mind changing it! I was looking at the different methods in python pertaining to dictionaries and I didn't see a "remove()" method, or any method that specifically deletes the key-value pair from the dictionary. They only had the "del" keyword for deleting the pairs, specific values, or the entire dictionary itself, so that's why I didn't use the remove method.